lib/improver/tests/utilities/cube_manipulation_new/test_MergeCubesForWeightedBlending.py

```diff
@@ -0,0 +1,308 @@
+        warning_msg = "model_id_attr not required"
+        plugin = MergeCubesForWeightedBlending(
+            "realization", model_id_attr="mosg__model_configuration")
+        self.assertTrue(any(item.category == UserWarning
+                            for item in warning_list))
+        self.assertTrue(any(warning_msg in str(item)
+                            for item in warning_list))
+        self.assertIsNone(plugin.model_id_attr)
+
+
+class Test__create_model_coordinates(IrisTest):
+    """Test the _create_model_coordinates method"""
+
+    def setUp(self):
+        """Set up some probability cubes from different models"""
+        data = np.array(
+            [0.9*np.ones((3, 3)), 0.5*np.ones((3, 3)), 0.1*np.ones((3, 3))],
+            dtype=np.float32)
+        thresholds = np.array([273., 275., 277.], dtype=np.float32)
+        time_point = dt(2015, 11, 23, 7)
+
+        # set up a MOGREPS-UK cube with 7 hour forecast period
+        self.cube_enuk = set_up_probability_cube(
+            data.copy(), thresholds, standard_grid_metadata='uk_ens',
+            time=time_point, frt=dt(2015, 11, 23, 0))
+
+        # set up a UKV cube with 4 hour forecast period
+        self.cube_ukv = set_up_probability_cube(
+            data.copy(), thresholds, standard_grid_metadata='uk_det',
+            time=time_point, frt=dt(2015, 11, 23, 3))
+
+        self.cubelist = iris.cube.CubeList([self.cube_enuk, self.cube_ukv])
+        self.plugin = MergeCubesForWeightedBlending(
+            "model", weighting_coord="forecast_period",
+            model_id_attr="mosg__model_configuration")
+
+    def test_basic(self):
+        """Test model ID and model configuration coords are created"""
+        self.plugin._create_model_coordinates(self.cubelist)
+        for cube in self.cubelist:
+            cube_coords = [coord.name() for coord in cube.coords()]
+            self.assertIn("model_id", cube_coords)
+            self.assertIn("model_configuration", cube_coords)
+            self.assertEqual(
+                cube.attributes["mosg__model_configuration"], "blend")
+
+    def test_null(self):
+        """Test no effect if model_id_attr is not set"""
+        plugin = MergeCubesForWeightedBlending("realization")
+        plugin._create_model_coordinates(self.cubelist)
+        for cube in self.cubelist:
+            cube_coords = [coord.name() for coord in cube.coords()]
+            self.assertNotIn("model_id", cube_coords)
+            self.assertNotIn("model_configuration", cube_coords)
+
+    def test_unmatched_model_id_attr(self):
+        """Test error if model_id_attr is not present on both input cubes"""
+        self.cubelist[0].attributes.pop("mosg__model_configuration")
+        msg = 'Cannot create model ID coordinate for grid blending '
+        with self.assertRaisesRegex(ValueError, msg):
+            self.plugin._create_model_coordinates(self.cubelist)
+
+
+class Test_process(IrisTest):
```

Comment:
If it helps.  My thought process is that there are three use cases for WeightedBlendAcrossWholeDimension:

- Cycle blending (over forecast period)
- Multi-model blending (over model_id or model_configuration)
- Collapsing realizations (via `improver-threshold` with the  `--collapse_coord` option)

So I've use those three blend coordinates in the "process" unit tests.  Collapsing realizations doesn't require any special setup, so that's just the one test to make sure `MergeCubes` is called without modifying the inputs.  For model blending I've tested model coordinate creation; and for both cycle and model blending I've tested `rationalise_blend_time_coordinates` indirectly, by setting up test cubes with different forecast periods and checking that the merged output has scalar `time` and `forecast_reference_time coordinates`.

Reply:
Thank you for explaining. I'm happy with this test coverage.